c++/src/kj/async-inl.h

```diff
@@ -2335,6 +2355,29 @@ class Coroutine<T>::Awaiter: public AwaiterBase {
   ExceptionOr<FixVoid<U>> result;
 };
 
+// ---------------------------------------------------------
+// Coroutine Magic
+//
+// KJ coroutines sometimes have access to "magic" functionality provided by coroutine adapter
+// implementations. To invoke this functionality, coroutines use the `co_magic` operator on a
+// "magic" object which the coroutine adapter implementation knows about.
+//
+// As of this writing, `kj::_::Coroutine<T>` does not itself provide any such magic functionality,
+// but soon will.
+
+#define co_magic co_yield
+// To invoke a coroutine's magic functionality `FOO`, use `co_magic FOO`. What `FOO` is, what the
+// expression evaluates to, and what side effects occur are all defined by the implementation of the
+// coroutine adapter.
+//
+// Coroutine magic is implemented in terms of the `co_yield` keyword, so this is just a define to
+// `co_yield`. The purpose of the macro is primarily as a visual signal that the code is not
+// actually yielding a value, but rather something different is going on.
+//
+// TODO(2.0): Unconditionally defining this is perhaps unwise. Should we have a KJ_DEFINE_CO_MAGIC
+//   macro to control this? Or we could `#define KJ_CO_MAGIC co_yield` and rely on projects to pass
```

Comment:
Maybe we should just have the macro be called `KJ_CO_MAGIC`. That tells readers that this is a KJ feature and hints that they can jump-to-definition for more info, rather than have them googling for it.

Reply:
Yeah, that works for me. The jump-to-definition is a great point.